functions.go

```diff
@@ -351,18 +351,21 @@ func Is(err, target error) bool {
 	return stderrors.Is(err, target)
 }
 
-// IsType is a convenience method for ascertaining if an error contains the
-// target error type within its chain. This is aimed at ease of development
-// where a more complicated error type wants to be to checked for existence but
-// pointer var of that type is too much overhead.
-func IsType[t error](err error) bool {
-	for err != nil {
-		if _, is := err.(t); is {
-			return true
-		}
-		err = stderrors.Unwrap(err)
-	}
-	return false
+// HasType is a convenience function around AsType there the assigned value
+// is discarded in favour of just the boolean answer. Use this when you don't
+// need the assigned value from AsType.
+
+// HasType is a function wrapper around AsType dropping the target return value
+// from AsType() making a function that can be used like this:
+//
+// return HasType[*MyError](err)
```

Comment:
FYI, if you indent these code lines with 1 space they'll be put in monospace / `<pre>` blocks in godoc output.